tests/resource/test_resource.py

```diff
@@ -147,3 +151,88 @@ def test_ipython_display(self, r, rep, capsys):
         r._ipython_display_()  # pylint: disable=protected-access
         captured = capsys.readouterr()
         assert rep in captured.out
+
+
+def _construct_tape_from_ops(lst_ops):
```

Comment:
Curious why this and some other function/classes are set as private? (the functions in test are kind of private anyway, no?)

Reply:
Yes! I must admit this is probably a result of the force of habit 😅! I don't think it matters if they are private or not.